sql/hive/src/test/scala/org/apache/spark/sql/hive/StatisticsSuite.scala

```diff
@@ -178,6 +179,7 @@ class StatisticsSuite extends QueryTest with BeforeAndAfterAll {
         sql(s"""SET ${SQLConf.AUTO_BROADCASTJOIN_THRESHOLD}=$tmp""")
       }
 
+      conf.setConf("spark.sql.autoSortMergeJoin", "true")
```

Comment:
I agree that we should add more unit test for sort merge join in `JoinSuite`, just be sure we generate the right physical plans.
In the meantime, for correctness testing, we can add an new class which extends the `HiveCompatibilitySuite`, and just  set the `spark.sql.autoSortMergeJoin` as `true` `false` in `beforeAll` and `afterAll` respectively.

What do you think?